ml_metadata/tools/mlmd_bench/fill_nodes_workload.cc

```diff
@@ -0,0 +1,321 @@
+  node_batch_name =
+      absl::StrCat("node_batch", absl::FormatTime(absl::Now()), "_", i);
+  num_properties =
+      GenerateRandomNumberFromUD(fill_nodes_config.num_properties(), gen);
+  string_value_bytes =
+      GenerateRandomNumberFromUD(fill_nodes_config.string_value_bytes(), gen);
+  num_nodes = GenerateRandomNumberFromUD(fill_nodes_config.num_nodes(), gen);
+  type_index = uniform_dist_type_index(gen);
+}
+
+// Gets the transferred bytes for certain `properties` of the current node.
+int64 GetTransferredBytesForNodeProperties(
+    const google::protobuf::Map<std::string, Value>& properties) {
+  int64 bytes = 0;
+  for (const auto& property : properties) {
+    // Includes the bytes for properties' name size.
+    bytes += property.first.size();
+    // Includes the bytes for properties' value size.
+    bytes += property.second.string_value().size();
+  }
+  return bytes;
+}
+
+// Calculates the transferred bytes for each node that will be inserted later.
+template <typename NT>
+int64 GetTransferredBytes(const NT& node) {
+  int64 bytes = 0;
+  // Increases `bytes` with the size of current node's `name` and
+  // `type_id`.
+  bytes += node.name().size() + 8;
+  // Increases `bytes` with the size of current node's `properties` and
+  // `custom_properties`.
+  bytes += GetTransferredBytesForNodeProperties(node.properties()) +
+           GetTransferredBytesForNodeProperties(node.custom_properties());
+  return bytes;
+}
+
+// Generates insert node.
+// For insert cases, the node's type will be `type` and its properties will be
+// generated w.r.t. `node_name`, `num_properties` and `string_value_bytes`.
+// Leaves it as return `void` instead of return `int64` for now because
+// FillNodes update mode will return `tensorflow::Status` in the future.
+template <typename T, typename NT>
+void GenerateNodes(const std::string& node_batch_name,
+                   const int64 num_properties, const int64 string_value_bytes,
+                   const T& type,
+                   google::protobuf::RepeatedPtrField<NT>& node_batch,
+                   int64& curr_bytes) {
+  CHECK((std::is_same<T, ArtifactType>::value ||
+         std::is_same<T, ExecutionType>::value ||
+         std::is_same<T, ContextType>::value))
+      << "Unexpected Types";
+  CHECK((std::is_same<NT, Artifact>::value ||
+         std::is_same<NT, Execution>::value ||
+         std::is_same<NT, Context>::value))
+      << "Unexpected Node Types";
+  // Insert nodes cases.
+  // Loops over all the node inside `node_batch` and sets up one by one.
+  for (int64 i = 0; i < node_batch.size(); ++i) {
+    node_batch[i].set_name(absl::StrCat(node_batch_name, "_node_", i));
+    node_batch[i].set_type_id(type.id());
+    // Uses "********" as the fake property value for current node.
+    std::string property_value(string_value_bytes, '*');
+    int64 curr_num_properties = 0;
+    // Loops over the types properties while generating the node's properties
+    // accordingly.
```

Comment:
int populated_num_properties = 0;
for (auto p : type.properties()) {
   // get p name, get p's value type  
   set_string_value / set_int_value / set_float_value
   node_batch[i].mutable_properties()[name].set...(...)
   if (++populated_num_properties > ..) {
       break;
   }
}

leave a TODO looks ok for now.